### PR TITLE
[MyRaft] [config] Creating protobuf config elements analogous to plugin.

### PR DESCRIPTION
**Pull request by iRitwik**
Creating protobuf config elements analogus to plugin. These will be passed along to opts and used by individual use cases - MB, commit rules, etc.

### CODE REVIEW

File: src/kudu/consensus/metadata.proto [hunk 1]
```diff
@@ -135,6 +175,16 @@ message RaftConfigPB {
 
   // The set of peers in the configuration.
   repeated RaftPeerPB peers = 3;
+
```
Comment (anirbanr-fb) [inline src/kudu/consensus/metadata.proto:208]: its probably not a good idea to keep configs which we are not going to get consensus on within RaftConfigPB
RaftConfigPB has an opid_index to mark its position in log and is stored in consensus metadata.
Its primary purpose is to have the ring members.

Comment (iRitwik) [reply src/kudu/consensus/metadata.proto:208]: This is exactly what I said before. How would you persist them?

Comment (iRitwik) [reply src/kudu/consensus/metadata.proto:208]: Repurposed the commit for creating analogues of thrift structs on plugin that will be populated in server options. Persistence shall be deferred.

File: src/kudu/consensus/metadata.proto [hunk 2]
```diff
@@ -135,6 +205,9 @@ message RaftConfigPB {
...
+  // Name of the replicaset.
+  optional string replicaset_name = 6;
```
Comment (anirbanr-fb) [inline src/kudu/consensus/metadata.proto]: replicaset_name does not make sense to be part of RaftPeerPB

File: src/kudu/consensus/metadata.proto [hunk 3]
```diff
@@ -71,6 +71,76 @@ message HealthReportPB {
...
+  // optional string master_ban_instance = 4;
+
+  // Name of the replicaset being configured: eg. mysql.replicaset.10001
+  optional string replicaset_name = 5;
+
+  // Server properties of the instance being configured.
+  optional RaftPeerPB server_config = 6;
+
+  // Replicaset topology information.
+  optional RaftTopologyPB raft_topology = 7;
```
Comment (anirbanr-fb) [inline src/kudu/consensus/metadata.proto:141]: I think we only need ReplicationTopologyPB here.

File: src/kudu/consensus/metadata.proto [hunk 4]
```diff
@@ -71,6 +71,76 @@ message HealthReportPB {
...
+
+  // Master ban to be
+  // optional bool master_ban_info = 3;
+  // optional string master_ban_instance = 4;
+
+  // Name of the replicaset being configured: eg. mysql.replicaset.10001
+  optional string replicaset_name = 5;
+
+  // Server properties of the instance being configured.
+  optional RaftPeerPB server_config = 6;
```
Comment (anirbanr-fb) [inline src/kudu/consensus/metadata.proto:138]: we probably don't need this here as we have a local_peer_pb_ inside simple_tablet_manager_
look at
void TSTabletManager::InitLocalRaftPeerPB()

File: src/kudu/consensus/metadata.proto [hunk 5]
```diff
@@ -71,6 +71,76 @@ message HealthReportPB {
...
+  // followers that exist per region
+  // This should include the replicas for the master, fallback regions and
+  // include LBU counts.
+  map<string, int32> follower_distribution = 3;
+}
+
+// Analogue of RaftTopology on the plugin.
+message RaftTopologyPB {
+  // Server properties of each member in the replicaset's raft ring.
+  repeated RaftPeerPB raft_server_properties = 1;
```
Comment (anirbanr-fb) [inline src/kudu/consensus/metadata.proto:119]: we probably dont need this here because RaftConfigPB already has the list of peers.

**Review by anirbanr-fb — APPROVED** [review]
Had a detailed offline conversation with iRitwik. 

we are using these protobufs to communicate requisite information to Raft.
However persistence of these protobufs might not happen on kuduraft side,
but on application side. We will revisit this soon.

**Pull request merged**